jupyter_server/utils.py

```diff
@@ -383,18 +383,15 @@ def _check_pid_posix(pid):
     check_pid = _check_pid_posix
 
 
-def maybe_future(obj):
-    """Like tornado's deprecated gen.maybe_future
-    but more compatible with asyncio for recent versions
-    of tornado
+def ensure_async(obj):
+    """Convert a non-async object to a coroutine if needed.
     """
-    if inspect.isawaitable(obj):
-        return asyncio.ensure_future(obj)
-    elif isinstance(obj, concurrent.futures.Future):
-        return asyncio.wrap_future(obj)
-    else:
-        # not awaitable, wrap scalar in future
-        f = asyncio.Future()
-        f.set_result(obj)
-        return f
+    if asyncio.iscoroutine(obj):
+        return obj
+    if type(obj) is asyncio.Future:
+        return obj
+
+    async def _():
+        return obj
 
+    return _()
```

Comment:
:+1: 

It might be worth one more set of eyes to specifically approve this change (maybe @kevin-bates). 

It LGTM, but I'm not super familiar with the async stuff. I want to make sure we're not missing any key edge cases that might lead to cryptic error messaging or bugs. Should we create a set of new tests specifically for this function?

Reply:
Yes, async is not easy, so any help would be appreciated here, including ways to test this function.